From b2413338adb9be469c0a5d27daa81cc49721aac9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 25 Sep 2024 13:59:25 +0200
Subject: [PATCH] feat(window): Add random color button to placeholder page

Starting with a random color is already possible with the Ctrl+R shortcut.
This change makes it more visible and accessible.

Also enlarge the window a bit to avoid scrolling with the new button.
---
 ...ithub.finefindus.eyedropper.gschema.xml.in |  4 +--
 data/resources/ui/window.blp                  | 34 ++++++++++++++-----
 2 files changed, 27 insertions(+), 11 deletions(-)

diff --git a/data/com.github.finefindus.eyedropper.gschema.xml.in b/data/com.github.finefindus.eyedropper.gschema.xml.in
index 39aed23..f2b2010 100644
--- a/data/com.github.finefindus.eyedropper.gschema.xml.in
+++ b/data/com.github.finefindus.eyedropper.gschema.xml.in
@@ -2,11 +2,11 @@
 <schemalist>
   <schema path="/com/github/finefindus/eyedropper/" id="@app-id@" gettext-domain="@gettext-package@">
     <key name="window-width" type="i">
-      <default>300</default>
+      <default>360</default>
       <summary>Window width</summary>
     </key>
     <key name="window-height" type="i">
-      <default>445</default>
+      <default>500</default>
       <summary>Window height</summary>
     </key>
     <key name="is-maximized" type="b">
diff --git a/data/resources/ui/window.blp b/data/resources/ui/window.blp
index accda2e..73aaf60 100644
--- a/data/resources/ui/window.blp
+++ b/data/resources/ui/window.blp
@@ -52,16 +52,32 @@ template $AppWindow: Adw.ApplicationWindow {
               title: _("No Color");
               description: _("Select a color from the screen to get started");
 
-              child: Button {
-                label: _("_Pick a Color");
-                use-underline: true;
-                halign: center;
-                action-name: "app.pick_color";
+              child: Box {
+                orientation: vertical;
+                spacing: 12;
 
-                styles [
-                  "pill",
-                  "suggested-action",
-                ]
+                Button {
+                  label: _("_Pick a Color");
+                  use-underline: true;
+                  halign: center;
+                  action-name: "app.pick_color";
+
+                  styles [
+                    "pill",
+                    "suggested-action",
+                  ]
+                }
+
+                Button {
+                  label: _("_Random Color");
+                  use-underline: true;
+                  halign: center;
+                  action-name: "app.random_color";
+
+                  styles [
+                    "pill",
+                  ]
+                }
               };
             };
           };
